src/Factory/FactoryClassGenerator.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-tooling for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-tooling/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Tooling\Factory;
+
+use ReflectionClass;
+use ReflectionParameter;
+
+class FactoryClassGenerator
+{
+    const FACTORY_TEMPLATE = <<<'EOT'
+<?php
+
+declare(strict_types=1);
+
+namespace %2$s;
+
+%3$s
+
+class %1$sFactory
+{
+    public function __invoke(ContainerInterface $container) : %1$s
+    {
+        return new %1$s(%4$s);
+    }
+}
+
+EOT;
+
+    public function createFactory(string $className) : string
+    {
+        $class = $this->getClassName($className);
+        $namespace = str_replace('\\' . $class, '', $className);
+        $constructorParameters = $this->getConstructorParameters($className);
+
+        $imports = array_keys($constructorParameters);
+        $imports[] = 'Psr\Container\ContainerInterface';
+
+        return sprintf(
+            self::FACTORY_TEMPLATE,
+            $class,
+            $namespace,
+            $this->formatImportStatements($imports),
+            $this->createArgumentString($constructorParameters)
+        );
+    }
+
+    private function getClassName(string $className) : string
+    {
+        return substr($className, strrpos($className, '\\') + 1);
+    }
+
+    private function getConstructorParameters(string $className) : array
+    {
+        $reflectionClass = new ReflectionClass($className);
+
+        if (! $reflectionClass || ! $reflectionClass->getConstructor()) {
```

Comment:
Hm... when we can have satisfied `! $reflectionClass`? I  can see that `ReflectionClass` can just throw exception on failure.

Reply:
You know, I have no idea. This was from the zend-servicemanager implementation, and, while I assume there was a reason for this, the PHP docs do indicate an exception is thrown normally. I'll update.